### PR TITLE
IIOD: Fix bug in mask bit tests (continued)

### PR DESCRIPTION
**Pull request by mhennerich**
From: #198 

Previously, the index field of struct iio_channel was being used to test
against the buffer field of struct iio_device and struct iio_buffer.
This is incorrect because the index of the first channel may be non-zero
and the indexes may not be contiguous. For further evidence, look at
iio_channel_enable/disable() which use number rather than index.

Signed-off-by: Michael Hennerich <michael.hennerich@analog.com>

### CODE REVIEW

File: iiod/ops.c [hunk 1]
```diff
@@ -560,7 +560,10 @@ static void rw_thd(struct thread_pool *pool, void *d)
 				struct iio_channel *chn = dev->channels[i];
 				long index = chn->index;
 
-				if (index >= 0 && TEST_BIT(entry->mask, i))
+				if (index < 0)
+					continue;
+
+				if (TEST_BIT(entry->mask, i))
```
Comment (commodo) [inline iiod/ops.c]: curios: should this be `chn->number` instead of `i`?

Comment (mhennerich) [reply iiod/ops.c]: The assumption is that both are the same! 
But Lars also thinks that chn->number looks more robust.
I'll fix this.

**Comment by commodo** [comment]
[LGTM](https://i.pinimg.com/564x/c7/89/d8/c789d8bbbf8b1ec2821369f5c8f0a58e.jpg)

**Comment by pcercuei** [comment]
Please split the libiio and IIOD changes into two commits.

**Comment by mhennerich** [comment]
> Please split the libiio and IIOD changes into two commits.

ok - Done!

**Pull request merged**